File: divisible-sum-pairs/divisible-sum-pairs.go
package divisiblesumpairs

func divisibleSumPairs(ignored int32, divisor int32, ns []int32) int32 {
	var result int32 = 0

	for i, n := range ns {
		for j := i + 1; j < len(ns); j++ {
			sum := n + ns[j]
			if sum%divisor == 0 {
				result++
			}
		}
	}

	return result
}

File: divisible-sum-pairs/DivisibleSumPairs.java
import java.util.List;

public class DivisibleSumPairs {

  public static int divisibleSumPairs(int ignored, int divisor, List<Integer> ns) {
    int result = 0;

    for (int i = 0; i < ns.size(); i++) {
      int n = ns.get(i);

      for (int j = i + 1; j < ns.size(); j++) {
        int sum = n + ns.get(j);
        if (sum % divisor == 0) {
          result++;
        } 
      }
    }

    return result;
  }

}
